### PR TITLE
{math}[foss/2019b] gmsh v4.5.6 w/ Python 3.7.4

### PR DESCRIPTION
**Pull request by jhein32**
config for gmsh 4.5.6 in foss 2019b and Python 3.7.4

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node3163.skitty.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/fe1d84239b032c66bacab55cd12dcf76 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2461.golett.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/29f7c11b4c7ce59dfb6ebbacdb80a9b8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, x86_64, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/4ee460dd23b518ea1e2985402d1bbdd2 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, x86_64, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/a973749200b756a5c1b88732a156ee04 for a full test report.

**Comment by jhein32** [comment]
@boegel How to fix that failed test?  I understand it passed on the other system?  We could just kick out upgrade of the old python 2.7 config to https.  That would solve the issue quickly, but isn't the most elegant.

**Comment by boegel** [comment]
@jhein32 I'll tackle it, no worries. Looks like it's picking up the wrong `python` when EasyBuild is using Python 3.x.

Probably not worth fixing in this old easyconfig (and it's a bug in the `CMakeLists.txt` of the software anyway).

**Comment by jhein32** [comment]
> @jhein32 I'll tackle it, no worries. Looks like it's picking up the wrong `python` when EasyBuild is using Python 3.x.
> 
> Probably not worth fixing in this old easyconfig (and it's a bug in the `CMakeLists.txt` of the software anyway).

Thanks.

**Comment by boegel** [comment]
@jhein32 Problem is fixed with the additional changes in https://github.com/jhein32/easybuild-easyconfigs/pull/6, please review/merge to update this PR...

File: easybuild/easyconfigs/g/gmsh/gmsh-3.0.6-foss-2017b-Python-2.7.14.eb [hunk 1]
```diff
@@ -19,15 +19,17 @@ builddependencies = [
...
 dependencies = [
-    ('Python', '2.7.14'),
+    ('Python', local_pymajmin + '.14'),
     ('PETSc', '3.8.3', '-downloaded-deps'),
     ('SLEPc', '3.8.3'),
 ]
 
 separate_build_dir = True
 
 configopts = '-DENABLE_FLTK=0 -DENABLE_WRAP_PYTHON=ON -DENABLE_METIS=1'
```
Comment (boegel) [inline easybuild/easyconfigs/g/gmsh/gmsh-3.0.6-foss-2017b-Python-2.7.14.eb:31]: @jhein32 Please add back the extra space at the end, it's needed to separate `-DENABLE_METIS=1` from `-DPython...`

Comment (jhein32) [reply easybuild/easyconfigs/g/gmsh/gmsh-3.0.6-foss-2017b-Python-2.7.14.eb]: Thanks.  I understood this and was doing it, while you were reviewing things.  I think all requests are done.  Let me know if there is stuff I overlooked.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, x86_64, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/27ad33a243d8a6afbfd5213773a56ded for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node3163.skitty.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/c2adaac4ea6b29b6fce722136cb48bde for a full test report.

**Comment by boegel** [comment]
Going in, thanks @jhein32!

**Pull request merged**